wrangler-service/src/main/java/io/cdap/wrangler/service/directive/WorkspaceHandler.java

```diff
@@ -437,6 +424,57 @@ schema, new Plugin("Wrangler", "transform", properties,
     });
   }
 
+  @POST
+  @TransactionPolicy(value = TransactionControl.EXPLICIT)
+  @Path("v2/contexts/{context}/workspaces/{id}/applyRecipe/{recipe-id}")
+  public void applyRecipe(HttpServiceRequest request, HttpServiceResponder responder,
+                          @PathParam("context") String namespace,
+                          @PathParam("id") String workspaceId,
+                          @PathParam("recipe-id") String recipeIdString) {
+    respond(responder, namespace, ns -> {
+      validateNamespace(ns, "Executing directives in system namespace is currently not supported");
+
+      RecipeId recipeId = RecipeId.builder(ns).setRecipeId(recipeIdString).build();
+      Recipe recipe = recipeStore.getRecipeById(recipeId);
+
+      DirectiveExecutionResponse response = execute(ns, request, new WorkspaceId(ns, workspaceId),
+                                                    recipe.getDirectives());
+      responder.sendJson(response);
+    });
```

Comment:
How do we send the errors like line number from applying recipe? Do you want to handle that in a separate task?

Reply:
Yes, we will be doing it in a separate PR.